app/domain/authentication/persist_auth_factory.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Authentication
+  # Allows creation of PersistAuth command class from an authenticator name
+  class PersistAuthFactory
+    class << self
+      def new_from_authenticator(authenticator)
+        Authentication::PersistAuth.new(
+          auth_initializer: auth_initializer(authenticator).new,
+          auth_data_class: auth_data_class(authenticator)
+        )
+      end
+
+      private
+
+      def auth_data_class(authenticator)
+        case authenticator
```

Comment:
With some slight renaming, we can greatly simplify this:

```ruby
def auth_data_class(authenticator)
   "Authentication::#{authenticator.underscore.camelize}::AuthenticatorData".classify
rescue 
  raise ArgumentError("AuthenticatorData class is not implemented for authenticator #{authenticator}")
end
```

**Note**: The above renames `K8sAuthenticatorData`/`AzureAuthenticatorData`/etc. to `AuthenticatorData`